src/main/java/com/zendesk/maxwell/schema/SchemaPosition.java

```diff
@@ -66,8 +66,9 @@ public void run() {
 			} catch (InterruptedException e) { }
 		}
 
-		if ( this.exception != null )
+		if ( exception == null ) {
```

Comment:
no, this isn't the main thread's exception, it's the position thread's exception.